### PR TITLE
Feat/rewards integration

### PR DESCRIPTION
**Pull request by pacmessica**
[notion](https://www.notion.so/figmentnetworks/b47c0f8b935741d8ac28717ddeea2038?v=5e4e1d3854094b788ae2b612329f27e8&p=cca7fea2cab5429895743ef7d28604b5)

todo - request structs are in `indexing-engine`, need to update go.mod to latest version after reward request structs are added [here](https://github.com/figment-networks/indexing-engine/pull/21)

### CODE REVIEW

File: manager/client/last.go [hunk 1]
```diff
@@ -253,6 +265,105 @@ func (hc *Client) LatestData(ctx context.Context, ldr shared.LatestDataRequest)
...
+			hc.logger.Warn("[CLIENT] still processing limit, retrying", zap.Time("rewards_time", ldr.RewardsTime), zap.Duration("since", time.Since(p.Started)))
+			hc.r.lock.Lock()
+			delete(hc.r.ProcessesRewardsData, key)
+			hc.r.lock.Unlock()
+		}
+
+		return ldResp, p.Error
+	} else {
+		hc.r.lock.Lock()
+		hc.r.ProcessesRewardsData[key] = ProcessRewardsData{
```
Comment (pacmessica) [inline manager/client/last.go:316]: set `ProcessesRewardsData` if doesnt exist

File: manager/client/last.go [hunk 2]
```diff
@@ -253,6 +265,105 @@ func (hc *Client) LatestData(ctx context.Context, ldr shared.LatestDataRequest)
...
+		}
+	}
+
+	hc.r.lock.Lock()
+	hc.r.ProcessesRewardsData[key] = ProcessRewardsData{
+		Finished:  true,
+		EndHeight: rewards.LastHeight,
+		LastTime:  rewards.LastTime,
+	}
+	hc.r.lock.Unlock()
```
Comment (pacmessica) [inline manager/client/last.go:362]: mark `ProcessRewardsData` as finished. Or should it be deleted at this point @lukanus ?

File: manager/client/last.go (hunk 1 above)
Comment (lukanus) [reply manager/client/last.go:316]: The line above inevitably returns. No need for else :)

**Review by lukanus — COMMENTED** [review]
lgtm :)

**Pull request merged**